File: LeetCode/Array/C++/PlusOne.cpp
#include <vector>
#include <iostream>
std::vector<int> plusOne(std::vector<int>& digits) {
    for (int i = digits.size() - 1; i >= 0; --i) {
        if (digits[i] == 9) {
            digits[i] = 0;
        } else {
            ++digits[i];
            return digits;
        }
    }
    digits[0] = 1;
    digits.emplace_back(0);
    return digits;
}

int main()
{
    std::vector<int> test1{1, 2, 3};
    std::vector<int> test2 = plusOne(test1);
    for(auto item : test1)
    {
        std::cout << item << " ";
    }
    std::cout << "\n";
    return 0;
}

File: LeetCode/Array/Python/PlusOne.py

def plus_one(digits):
	for i in reversed(range(len(digits))):
		if digits[i] == 0:
			digits[i] = 0
		else:
			digits[i] += 1
			return digits
	digits[0] = 1
	digits.append(0)
	return digits

def main():
	test1 = [1, 2, 3]
	test1 = plus_one(test1)
	for digit in test1:
		print(digit)
	#print("\n")

if __name__ == "__main__":
	main()